app/employee-app/backend/modules/database/db_functions.bal

```diff
@@ -78,3 +79,111 @@ public isolated function getEmployees(EmployeeFilter filters, int 'limit, int of
             lead: dbEmployeeInfo.lead == 1
         };
 }
+
+# Retrieve organizational details from HRIS.
+#
+# + employeeStatuses - List of employee statuses to consider
+# + return - List of business units
+public isolated function getOrgStructure(string[]? employeeStatuses) returns OrgDataResponse|error {
+
+    OrgData[] orgDataDBResponse = [];
+    stream<OrgDataDB, sql:Error?> resultStream = databaseClient->query(getOrgStructureQuery(employeeStatuses));
+    error? iterateError = from OrgDataDB orgData in resultStream
```

Comment:
`iterateError` naming is not correct. This won't be an `error` always.
You can use something like `orgResult`.